src/datasource/components/ItemCountWarning.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { css } from '@emotion/css';
+import { GrafanaTheme2 } from '@grafana/data';
+import { Icon, useStyles2 } from '@grafana/ui';
+import { ITEM_COUNT_WARNING_THRESHOLD } from '../constants';
+
+interface ItemCountWarningProps {
+  itemCount: number;
+  threshold?: number;
+}
+
+/**
+ * A warning banner that displays when the query matches too many items.
+ * This is a non-intrusive warning that doesn't block the query flow.
+ */
+export const ItemCountWarning: React.FC<ItemCountWarningProps> = ({
+  itemCount,
+  threshold = ITEM_COUNT_WARNING_THRESHOLD,
+}) => {
+  const styles = useStyles2(getStyles);
+
+  if (itemCount < threshold) {
+    return null;
+  }
+
+  return (
+    <div className={styles.warningContainer}>
```

Comment:
Should we use `grafana.ui`'s `Alert` here as well to avoid doing custom styling ourselves?